webapp/src/CodeSplitComponent.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component, ComponentType } from "react"
+
+type Props = any
+
+interface State {
+  component: ComponentType | null
+}
+
+export const codeSplit = (importComponent: () => Promise<any>, importKey: string) => {
+  class AsyncComponent extends Component<Props, State> {
+    constructor(props: Props) {
+      super(props)
+
+      this.state = {
+        component: null
+      }
+    }
+
+    public async componentDidMount() {
+      const importedModule = await importComponent()
```

Comment:
We should probably render some error condition in these cases. It's effectively a failed network request, and mobile users will run into this issue a lot. I know we have those error snackbars that pop up on failed renders, so we can act on a try/catch here.

Else, the error will bubble up and the entire application will crash on these pages.

Reply:
You are right, could handle it like that:
- redirect to HomePage 
- show an error notification that the page failed to load
will this be ok?

Reply:
The error notification is fine but I'm a little hesitant to redirect since that causes the user to make additional network requests (and they already have a shoddy connection since they're seeing the error). Perhaps we could just show something simple in place of the component like a `<Typography>Error loading this component. Please check your network connection and reload this page</Typography>`

Reply:
Ok, my reasoning was that if they're loading additional chunks, they probably came from the main page and it's loaded, so redirecting it inside react router would not incur additional requests.

I'll do a plain-html message today, like you suggested :+1: